### PR TITLE
feat(onebot): support file message

### LINKED ISSUE #109: Feat: support onebot file receive message
### description

It seems the current onebot platform adapter may not be handling `private offline files` or `group files` received its way. I believe an improvement would be for the onebot adapter to have the ability to receive and process those kinds of message.

### PR DESCRIPTION
**Pull request by Seidko**
Support group file upload and offline file recived.

This pull request may have some issues. I would appreciate any feedback to help improve it. 🙏

close #109

### CODE REVIEW

File: adapters/onebot/src/utils.ts [hunk 1]
```diff
@@ -252,6 +249,20 @@ export async function adaptSession(bot: BaseBot, data: OneBot.Payload) {
         session.type = 'onebot'
         session.subtype = 'channel-destroyed'
         break
+      case 'offline_file':
+        session.elements = [h('file', data.file)]
+        session.content = session.elements.join('')
```
Comment (shigma) [inline adapters/onebot/src/utils.ts:254]:
```suggestion
```

这一行是不需要的应该。

File: adapters/onebot/src/utils.ts [hunk 2]
```diff
@@ -252,6 +249,20 @@ export async function adaptSession(bot: BaseBot, data: OneBot.Payload) {
...
+        session.type = 'message'
+        session.subtype = 'private'
+        session.subsubtype = 'offline-file-added'
+        break
+      case 'group_upload':
+        session.elements = [h('file', data.file)]
+        session.content = session.elements.join('')
```
Comment (shigma) [inline adapters/onebot/src/utils.ts:261]:
```suggestion
```

File: adapters/onebot/src/utils.ts [hunk 3]
```diff
@@ -252,6 +249,18 @@ export async function adaptSession(bot: BaseBot, data: OneBot.Payload) {
         session.type = 'onebot'
         session.subtype = 'channel-destroyed'
         break
+      case 'offline_file':
+        session.elements = [h('file', data.file)]
```
Comment (shigma) [inline adapters/onebot/src/utils.ts:253]:
```suggestion
        session.elements = [h.file(data.file)]
```

h 二参数应该是 attr 或者 children，你这个 data.file 是 string 吧？

**Review by shigma — CHANGES_REQUESTED** [review]

File: adapters/onebot/src/utils.ts (hunk 3 above)
Comment (Seidko) [reply adapters/onebot/src/utils.ts:253]: `data.file` is object, [here is it's type](https://github.com/Seidko/satori/blob/17d1e7bad3e3d6f09310efe13139eed746d394c6/adapters/onebot/src/types.ts#L401C1-L408):
```ts
export interface Payload extends Message {
  // ...
  file: File
}

export interface File {
  name: string
  size: number
  url: string
}
```

**Review by shigma — APPROVED** [review]

**Pull request merged**